main.tf

```diff
@@ -103,102 +105,71 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
   }
 
   dynamic "default_cache_behavior" {
-    for_each = [for i in var.dynamic_default_cache_behavior : {
-      allowed_methods             = i.allowed_methods
-      cached_methods              = i.cached_methods
-      target_origin_id            = i.target_origin_id
-      compress                    = i.compress
-      query_string                = i.query_string
-      cookies_forward             = i.cookies_forward
-      headers                     = i.headers
-      viewer_protocol_policy      = i.viewer_protocol_policy
-      min_ttl                     = i.min_ttl
-      default_ttl                 = i.default_ttl
-      max_ttl                     = i.max_ttl
-      lambda_function_association = lookup(i, "lambda_function_association", null)
-    }]
+    for_each = var.dynamic_default_cache_behavior[*]
+
     content {
       allowed_methods  = default_cache_behavior.value.allowed_methods
       cached_methods   = default_cache_behavior.value.cached_methods
       target_origin_id = default_cache_behavior.value.target_origin_id
-      compress         = default_cache_behavior.value.compress
+      compress         = lookup(default_cache_behavior.value, "compress", null)
 
       forwarded_values {
         query_string = default_cache_behavior.value.query_string
         cookies {
           forward = default_cache_behavior.value.cookies_forward
         }
-        headers = default_cache_behavior.value.headers
+        headers = lookup(default_cache_behavior.value, "headers", null)
       }
 
       dynamic "lambda_function_association" {
-        for_each = default_cache_behavior.value.lambda_function_association == null ? [] : [for i in default_cache_behavior.value.lambda_function_association : {
```

Comment:
This line should have already made the `lambda_function_association` optional.  The new syntax is much simpler but that feature should have already existed

Reply:
it hasn't, tested without it but it was still expecting the argument

Reply:
Ok ya I was pretty sure it was working as expected with the old syntax.  The new syntax makes it more concise, so if that is working no issues